### PR TITLE
🐛 Filter out `search` from query when using subcommands

### PR DESCRIPTION
**Pull request by segersniels**
## Description

With the introduction of the subcommand when you use `gitmoji search bug linter` it will always include the `search` keyword in the query which is a regression. This PR ensures the `search` input is filtered out of `cli.input` when the command is executed through a subcommand.

It changes how the `search` command is handled. Instead of going over each input from within `cli.js` we now determine it in `findGitmojiCommand` and pass the correct option to the command depending on the situation. Where it is then further handled by the `search` command itself.

## Tests

- [x] All tests passed.

### CODE REVIEW

**Comment by segersniels** [comment]
@carloscuesta probably good to get this one merged at one point as well since there's a regression when searching through the subcommand

File: src/utils/findGitmojiCommand.js [hunk 1]
```diff
@@ -6,34 +6,67 @@ const isSupportedCommand = (command: ?string, options: Object): boolean => {
...
+
+  if (!command) {
+    type = 'command'
+    command = input[0]
+  }
+
+  return {
+    type,
+    command
+  }
```
Comment (carloscuesta) [inline src/utils/findGitmojiCommand.js:30]: Let's refactor this into:

```suggestion
  const command = Object.keys(flags)
    .map((flag) => flags[flag] && flag)
    .find((flag) => options[flag])

  return command
    ? {
      type: 'flag',
      command
    }
    : {
      type: 'command',
      command: input[0]
    }
```

**Review by carloscuesta — COMMENTED** [review]
Sorry for the delay!

Thanks for the PR 🙏🏼

File: src/commands/search/index.js [hunk 2]
```diff
@@ -3,10 +3,21 @@ import filterGitmojis from '@utils/filterGitmojis'
...
+  query?: string[]
+}
+
+const search = (options: SearchOptions): Promise<void> => {
   return getEmojis()
-    .then((gitmojis) => filterGitmojis(query, gitmojis))
-    .then((gitmojisFiltered) => printEmojis(gitmojisFiltered))
+    .then((gitmojis) => {
+      if (!options.query) {
+        return printEmojis(gitmojis)
```
Comment (carloscuesta) [inline src/commands/search/index.js:14]: Hey! Can we add a test case for this?

Comment (segersniels) [reply src/commands/search/index.js:14]: Oh sorry, I completely forgot this PR and comment. Sorry for the delay!

**Pull request merged**